### PR TITLE
Add styling documentation

### PR DESCRIPTION
**Pull request by TimLariviere**
Part of #927

### CODE REVIEW

File: docs/content/en/v2/development/styling.md [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+```fs
+[<Extension>]
+static member inline SomeExtensionMethod(this: TypeForWhichWeWantToAddAnExtension) =
+```
+
+**Now, the most important part to understand:** when using a widget, Fabulous will actually give you an instance of `WidgetBuilder<'msg, 'marker>` where `'marker` will be a type representing the widget.  
+For example, `Label()` will give you `WidgetBuilder<'msg, ILabel>`,
+
+You might have noticed that in the code above, we are using `#ILabel` instead of `ILabel`.  
+The `#` (pound) symbol lets F# know that we actually want the shorthand modifier to be available to all widgets deriving from `ILabel`, not just `Label`.
```
Comment (edgarfgp) [inline docs/content/en/v2/development/styling.md:154]: Should we add a link the the Flexible types feature ? https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/flexible-types

**Review by edgarfgp — COMMENTED** [review]
Should we also mention that there is the .resources modifier that will apply global styling ?

**Comment by TimLariviere** [comment]
Hmm yes, you're right. I should add the "legacy" way to do styling via `.resources()`, `.style()`

File: docs/content/en/v2/development/styling.md (hunk 1 above)
Comment (TimLariviere) [reply docs/content/en/v2/development/styling.md]: Good idea! Didn't even know that it was called a flexible type 😅

**Pull request merged**